SofaKernel/modules/Sofa.Config/src/sofa/config.h.in

```diff
@@ -123,31 +123,40 @@ typedef unsigned __int64	uint64_t;
 #  define SOFA_PRAGMA_MESSAGE(text) SOFA_DO_PRAGMA(message #text)
 #  define SOFA_PRAGMA_WARNING(text) SOFA_DO_PRAGMA(GCC warning #text)
 #endif
+
+/************* DEPRECATION MACROS *************/
+
 #define SOFA_DEPRECATED_HEADER(untilVersion, newHeader) SOFA_PRAGMA_WARNING( \
     This header is deprecated and will be removed at SOFA untilVersion.      \
     To fix this warning you must include newHeader instead. )
 
-#define SOFA_DEPRECATE_ASPECT(toFixMsg) \
-    [[deprecated(" \
-    Aspect has been deprecated since 2020-03-25 (see PR 1269). \
-    Related functions have been removed since 2021-03-25 (see PR 1793). " toFixMsg)]]
-
-#define SOFA_DEPRECATE_BASEDATA_API(toFixMsg) \
-    [[deprecated("BaseData Api has changed since 2021-01-01 (see PR 1503)." toFixMsg)]]
-
-#define SOFA_DEPRECATE_DATA_API(toFixMsg) \
-    [[deprecated("Data Api has changed since 2021-01-01 (see PR 1753)." toFixMsg)]]
-
-#define SOFA_DEPRECATE_DATALINK(toFixMsg) \
-    [[deprecated("Link Api has changed since 2020-10-03 (see PR 1503)." toFixMsg)]]
-
-#define SOFA_DEPRECATE_COMPONENTSTATE(toFixMsg) \
-    [[deprecated("The attribute was renamed to d_componentState in PR#1358. " toFixMsg)]]
-
-#define SOFA_DEPRECATE_CLASSNAME_INTROSPECTION(toFixMsg) \
-    [[deprecated("This function has been deprecated in #PR 1283. The function will be removed " \
-                 "the 01.01.2021. Information on how to update your code is provided in the PR " \
-                 "description. Information on how to update your code is provided in the PR description.")]]
+#define SOFA_ATTRIBUTE_DEPRECATED(deprecateDate, removeDate, toFixMsg) \
+    [[deprecated( \
+    "It is still usable but has been DEPRECATED since " deprecateDate ". " \
+    "Removal is planned for " removeDate ". " toFixMsg)]]
+
+#define SOFA_ATTRIBUTE_DISABLED(deprecateDate, disableDate, toFixMsg) \
+    [[deprecated( \
+    "It is not usable anymore because it has been DISABLED since " disableDate " " \
+    "(after being deprecated on " deprecateDate "). " toFixMsg)]]
+
+#define SOFA_ATTRIBUTE_DEPRECATED_EXECPARAMS(msg) \
```

Comment:
I found the name misleading, as the changes are supposed to be part of the ASPECT deprecation flow (i.e. not independant).